Sources/JWTKit/X5CVerifier.swift

```diff
@@ -121,13 +121,27 @@ public struct X5CVerifier: Sendable {
             return data
         }
 
-        // Setup an untrusted chain using all the certificates in the x5c.
+        // Setup an untrusted chain using all the certificates in the x5c
         let untrustedChain = try CertificateStore(certificateData.map {
             try Certificate(derEncoded: [UInt8]($0))
         })
 
+        var date = Date()
+        var decoder = jsonDecoder
+        if Payload.self is ValidationTimePayload.Type {
+            decoder.dateDecodingStrategy = .millisecondsSince1970
+        }
+
+        let payload = try parser.payload(as: Payload.self, jsonDecoder: decoder)
+
+        if let validationTimePayload = payload as? ValidationTimePayload {
```

Comment:
I wonder if we could check the against the type itself rather than the struct to avoid a double parse